src/einsteinpy/utils.py

```diff
@@ -51,13 +52,53 @@ def scalar_factor(era, t, tuning_param=1):
     ValueError : If era is not 'md' , 'rd', and 'ded'.
 
     """
+    T = t.to(u.s).value
     if era == "md":
-        return tuning_param * (t ** (2 / 3))
+        return tuning_param * (T ** (2 / 3))
     elif era == "rd":
-        return tuning_param * (t ** (0.5))
+        return tuning_param * (T ** (0.5))
     elif era == "ded":
         hubble_const = (constant.Cosmo_Const / 3) ** 0.5
-        val = np.e ** (hubble_const * t)
+        val = np.e ** (hubble_const.value * T)
+        return tuning_param * val
+    else:
+        raise ValueError("Passed era is out of syllabus!")
```

Comment:
@ritzvik  How about formalizing this one? Maybe this will be too much for professional users.